### PR TITLE
[Enhancement]Minor improvements

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🎯 Goal

Add comments, resolve warnings and improve logging.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (Docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamVideo/WebRTC/WebRTCClient.swift [hunk 1]
```diff
@@ -1311,9 +1311,10 @@ class WebRTCClient: NSObject, @unchecked Sendable {
     
     private func addOnParticipantsChangeHandler() {
         Task { [weak self] in
```
Comment (laevandus) [inline Sources/StreamVideo/WebRTC/WebRTCClient.swift:1313]: Unrelated but Is it OK that the Task is never cancelled?

**Review by laevandus — APPROVED** [review]

File: Sources/StreamVideo/WebRTC/WebRTCClient.swift [hunk 2]
```diff
@@ -1311,9 +1311,10 @@ class WebRTCClient: NSObject, @unchecked Sendable {
...
+            guard let self else { return }
```
Comment (martinmitrevski) [inline Sources/StreamVideo/WebRTC/WebRTCClient.swift:1314]: I think I already fixed this in another PR.

Comment (ipavlidakis) [reply Sources/StreamVideo/WebRTC/WebRTCClient.swift:1319]: I didn't find somewhere where we are actually terminating it so adding the fix here.

File: Sources/StreamVideo/WebRTC/WebRTCClient.swift (hunk 1 above)
Comment (ipavlidakis) [reply Sources/StreamVideo/WebRTC/WebRTCClient.swift]: Definitely! Thanks :)

**Review by martinmitrevski — APPROVED** [review]

**Pull request merged**